packages/std/src/testing/mock.rs

```diff
@@ -114,12 +129,21 @@ pub struct MockApi {
     /// Length of canonical addresses created with this API. Contracts should not make any assumptions
     /// what this value is.
     canonical_length: usize,
+    /// Prefix used for creating addresses in Bech32 encoding.
+    bech32_prefix: &'static str,
+    /// Variant of the Bech32 encoding used for generating addresses.
+    bech32_variant: Variant,
+    /// Maximum length of the inout data used for generating Bech32 encoded addresses.
+    bech32_input_length: usize,
```

Comment:
Do we need this flexibility? We can just generate 32 byte addresses in all cases IMO.

Reply:
In multi-test, the default `Api` trait implementation uses `cosmwasm_std::testing::MockApi`. This makes sense, while users become the same behavior in unit tests and in tests written using multi-test.

To make `MockApi` usable in multi-test it should be IMO maximally configurable (option A).

Another approach is, that we stick to Bech32 version with 32 bytes in `MockApi` and provide more flexible implementation in multi-test (option B).

Which option should be applied here?

Reply:
Right, but why would a user care if the generated `Addr`s are 20 or 32 bytes? All contract code should be agnostic to that.

Reply:
Ok, provided simplified implementation.